.github/workflows/docker-image.yml

```diff
@@ -0,0 +1,45 @@
+name: "Docker Image"
+
+on:
+  push:
+    branches:
+      - "main"
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
```

Comment:
Can you please use a specific version (`v3.0.0`, for example), instead of a commit? It makes it a bit more readable, if that makes sense.